src/utils/data/ManagedHash.ts

```diff
@@ -0,0 +1,71 @@
+import { type GCable, type GCData } from '../../rendering/renderers/shared/GCSystem';
+import { type Renderer } from '../../rendering/renderers/types';
+
+import type EventEmitter from 'eventemitter3';
+
+/**
+ * A hash for managing renderable and resource resources with GC integration.
+ * @internal
+ */
+export class ManagedHash<T extends GCable & { uid: number } & Pick<EventEmitter, 'once' | 'off'>>
+{
+    // Exposed directly for GC system access
+    public items: Record<number, T> = Object.create(null);
+    private _renderer: Renderer;
```

Comment:
small - but we should prolly just pass in the gc system here rather than the renderer as thats the only bit we need.